### PR TITLE
fix: validate endpointOverride URI scheme (#626)

### LINKED ISSUE #626: Endpoint override type is not strict enough
Hi!
I got a strange exception trying to connect s3 client to local minio.

```
software.amazon.awssdk.core.exception.SdkClientException: Unable to marshall request to JSON: baseUri must not be null.
        at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:98) ~[software.amazon.awssdk.sdk-core-2.17.61.jar:?]
        at software.amazon.awssdk.services.s3.transform.HeadObjectRequestMarshaller.marshall(HeadObjectRequestMarshaller.java:53) ~[software.amazon.awssdk.s3-2.17.61.jar:?]
        ...
        at io.github.vigoo.zioaws.s3.package$S3Impl.$anonfun$headObject$1(package.scala:2392) ~[io.github.vigoo.zio-aws-s3_2.13-3.17.61.1.jar:3.17.61.1]

Caused by: java.lang.NullPointerException: baseUri must not be null.
        at software.amazon.awssdk.utils.Validate.paramNotNull(Validate.java:156) ~[software.amazon.awssdk.utils-2.17.61.jar:?]
        ...
```

After some experiments, I found that the type of `endpointOverride` defined here: https://github.com/vigoo/zio-aws/blob/master/zio-aws-core/src/main/scala/io/github/vigoo/zioaws/core/config/package.scala#L237-L239 allows for incorrect URIs to be passed to underlying aws sdk.

In my case it was `minio:9000` instead of correct `http://minio:9000`

In my semi-educated guess, it would be better to change `endpointOverride` descriptor to `url` to enforce type safety.

### PR DESCRIPTION
**Pull request by darkenpeng**
## Summary                                                         
Add `httpUri` config descriptor that validates the URI scheme is `http` or `https`. This prevents runtime `NullPointerException` when invalid URIs (e.g., `minio:9000` without scheme) are passed to AWS SDK.

### Problem
When configuring `endpointOverride` with an invalid URI like `minio:9000` (missing scheme), the config parsing succeeds but AWS SDK throws NPE at runtime.                                          

### Solution                                                                      
  - Add `httpUri` descriptor that validates scheme is `http` or `https`
  - Provide clear error message at config time

### Changes
- `descriptors/package.scala`: Add `httpUri` descriptor with scheme validation
  - `CommonAwsConfigSpec.scala`: Add tests for valid/invalid endpoints 

Fixes #626

### CODE REVIEW

**Comment by darkenpeng** [comment]
About the edge case discussion in #626
K8s proxy scenarios needing URI flexibility. I looked into this:
| Use Case | Endpoint | Works? |
|----------|----------|--------|
| LocalStack | http://localhost:4566 | ✅ |
| MinIO | http://minio:9000 | ✅ |
| K8s service | http://my-proxy.namespace:8080 | ✅ |
| AWS | https://s3.amazonaws.com | ✅ |

All real-world cases I could find use http:// or https://. Makes sense since AWS SDK communicates over HTTP(S).
Also, AWS SDK 2.x [explicitly requires the scheme](https://github.com/aws/aws-sdk-java-v2/issues/5831) - it's an intentional breaking change from v1.x. The [official docs](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/endpoint-config.html) always show full URLs with schemes too.

File: zio-aws-core/src/main/scala/zio/aws/core/config/descriptors/package.scala [hunk 1]
```diff
@@ -4,9 +4,28 @@ import software.amazon.awssdk.auth.credentials._
...
+   * A URI config that validates the scheme is http or https.
+   * This prevents runtime NullPointerExceptions when invalid URIs
+   * (e.g., "minio:9000" without scheme) are passed to AWS SDK.
+   *
+   * @see https://github.com/zio/zio-aws/issues/626
+   */
+  val httpUri: Config[URI] = Config.uri.mapAttempt { uri =>
+    val scheme = uri.getScheme
+    if (scheme == null || (!scheme.equalsIgnoreCase("http") && !scheme.equalsIgnoreCase("https"))) {
+      throw new IllegalArgumentException(
```
Comment (guizmaii) [inline zio-aws-core/src/main/scala/zio/aws/core/config/descriptors/package.scala]: What's the type signature of `mapAttempt`. please?

I'm pretty sure it take a function `URI => Either[String, A]` and I'm pretty sure you're not supposed to throw

Comment (darkenpeng) [reply zio-aws-core/src/main/scala/zio/aws/core/config/descriptors/package.scala]: Thanks for the quick review! You're right — I've updated it to use `mapOrFail` with explicit Either instead.

**Review by guizmaii — APPROVED** [review]
LGTM! Thanks @darkenpeng :)
